### ISSUE TITLE
Unexpected outputs of randomly initialized `T5ForConditionalGeneration`.

### ISSUE THREAD

**Issue by Tomiinek** [huggingface/transformers#15007]
## Environment info

- `transformers` version: 4.4.2
- Platform: Linux-5.11.0-43-generic-x86_64-with-glibc2.29
- Python version: 3.8.10
- PyTorch version (GPU?): 1.6.0 (False)

### Who can help
@patrickvonplaten

## Information

If I instantiate `T5ForConditionalGeneration` from config, it outputs unexpected logits leading to ground-truth tokens!
If instantiating the model with pre-trained weights, i.e. `T5ForConditionalGeneration.from_pretrained('t5-small')`, outputs seem to be ok.

I am might be doing something illegal, but it is not apparent to me from the documentation. I am also not sure how to set `decoder_attention_mask`. I think that we do not have to care about it, but see [this question](https://discuss.huggingface.co/t/decoder-attention-mask-in-text2text-se2seq-generation-encoder-decoder-models/5851).

## To reproduce

``` python
import torch
from transformers import AutoConfig, T5ForConditionalGeneration

config = AutoConfig.from_pretrained('t5-small')
model = T5ForConditionalGeneration(config)

decoder_input_ids = torch.tensor([[0, 2, 3, 1, 0]])
decoder_attention_mask = torch.tensor([[ True,  True,  True,  True,  False]])

model.eval()
outputs = model(   
    input_ids=torch.zeros(bs, 1, device=dev).long(),       # dummy encoder input
    attention_mask=torch.ones(bs, 1, device=dev).float(),  # dummy encoder input mask
    decoder_input_ids=decoder_input_ids,
    decoder_attention_mask=decoder_attention_mask
)

print(outputs.logits.argmax(-1)) # outputs tensor([[0, 2, 3, 1, 0]])

model.train()
outputs = model(   
    input_ids=torch.zeros(bs, 1, device=dev).long(),       # dummy encoder input
    attention_mask=torch.ones(bs, 1, device=dev).float(),  # dummy encoder input mask
    decoder_input_ids=decoder_input_ids,
    decoder_attention_mask=decoder_attention_mask
)

print(outputs.logits.argmax(-1)) # outputs tensor([[10526,  2, 3, 1, 30997]])
```

## Expected behavior

I expect some random tokens such as `tensor([[123, 42, 666, 73, 13]])` instead of the exact copy of input. The model is not trained at all and moreover should be causal. The pre-trained model outputs `tensor([[0, 0, 0, 0, 0]])` in both cases.
